hongit-api/src/main/kotlin/com/devfloor/hongit/api/article/application/ArticleService.kt

```diff
@@ -111,18 +154,51 @@ class ArticleService(
     }
 
     @Transactional(readOnly = true)
-    fun showAllByNickname(nickname: String): List<ArticleFeedResponse> {
-        val articles = userRepository.findByNicknameOrNull(nickname)
-            ?.let(articleRepository::findAllByAuthor)
+    fun showAllByNickname(nickname: String, page: Int, pageSize: Int, loginUser: User): List<ArticleFeedResponse> {
+        val user = userRepository.findByNicknameOrNull(nickname)
             ?: EntityNotFoundException.notExistsNickname(User::class, nickname)
 
-        return articles
-            .filter { !it.anonymous }
+        val articles = articleRepository.findAllByAuthor(user, PageRequest.of(page, pageSize))
+        if (loginUser.id != user.id) {
+            return articles.content.filter { !it.anonymous }
+                .map {
+                    ArticleFeedResponse(
+                        article = it,
+                        articleOptions = articleOptionRepository.findAllByArticle(it),
+                        articleFavorites = articleFavoriteRepository.findAllByArticle(it),
+                        page = page,
+                        totalArticleCount = articleRepository.countAllByAuthor(user)
+                    )
+                }
+        } else {
+            return articles.content
+                .map {
+                    ArticleFeedResponse(
+                        article = it,
+                        articleOptions = articleOptionRepository.findAllByArticle(it),
+                        articleFavorites = articleFavoriteRepository.findAllByArticle(it),
+                        page = page,
+                        totalArticleCount = articleRepository.countAllByAuthor(user)
+                    )
+                }
+        }
```

Comment:
```suggestion
        return articles.content
            .filter { loginUser.hasSameId(user) || !it.anonymous } // 내부 predicate 로직을 따로 메서드로 뽑아도 좋고
            .map {
                ArticleFeedResponse(
                    article = it,
                    articleOptions = articleOptionRepository.findAllByArticle(it),
                    articleFavorites = articleFavoriteRepository.findAllByArticle(it),
                    page = page,
                    totalArticleCount = articleRepository.countAllByAuthor(user)
                )
            }
```

이렇게 수정하면 좀 더 펀.쿨.섹시해 보이는데 어떤지여??